lib/active_record_shards/connection_specification.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
-ActiveRecord::Base.class_eval do
-  def self.establish_connection(spec = ENV["DATABASE_URL"])
+class << ActiveRecord::Base
+  remove_method :establish_connection unless ActiveRecord::VERSION::MAJOR == 4
```

Comment:
Why the `unless`?

Reply:
I wish we could just use `Module#prepend` instead of overwriting everything…

Reply:
Apparently we're getting rid of newrelic so.... might solve our problem...

Reply:
can do ...

Reply:
hmm here it makes little sense since it's a class method and we don't call super ... so would add more indentation/weirdness ...

Reply:
Won't be able to use Module.prepend while we still have `alias` around

Reply:
I still don't understand the `unless`. This method is redefined for all Rails versions, right?

Reply:
Mail threading:
> defined on rails 3 and 5 ... but not 4 ...

Reply:
It’s also defined in Rails 4: https://github.com/rails/rails/blob/v4.2.0/activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb#L599